presto-main/src/main/java/com/facebook/presto/server/ResourceGroupStateInfoResource.java

```diff
@@ -60,10 +68,29 @@
 @RolesAllowed(ADMIN)
 public class ResourceGroupStateInfoResource
 {
+    private static class ResourceGroupStateInfoKey
```

Comment:
It seems we would need a hash function for key match?

Reply:
yeah we need` hashCode()` and `equals() `methods here